src/vehicle_move.cpp

```diff
@@ -2139,23 +2139,23 @@ float map::vehicle_wheel_traction( const vehicle &veh, bool ignore_movement_modi
             continue; // Ignore the movement modifier if caller specifies a bool
         }
 
+        int wheel_ter_mod = 0;
         for( const veh_ter_mod &mod : vpi.wheel_info->terrain_modifiers ) {
             const bool ter_has_flag = tr.has_flag( mod.terrain_flag );
             if( ter_has_flag && mod.move_override ) {
-                move_mod = mod.move_override;
+                wheel_ter_mod = mod.move_override;
                 break;
             } else if( !ter_has_flag && mod.move_penalty ) {
-                move_mod += mod.move_penalty;
-                break;
+                wheel_ter_mod += mod.move_penalty;
             }
         }
```

Comment:
Did you mean to remove the `break` in `else if`, but still keep it in `if`?

I don't understand the code, it just doesn't seem right.